cmake/extensions.cmake

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# cddl_source(<target> <cddl_file> ENTRY_TYPES <entry_types> [VERBOSE])
+#
+# Add generated code to the project for decoding CBOR.
+#
+# <cddl_file> is a file with CDDL data describing the expected CBOR data.
+#
+# <entry_types> is a list of the types defined in the CDDL file for which there
+#               should be an exposed decoding function.
+#
+# Specify VERBOSE to add printing to the generated code. Check kconfig to add
+# printing in the low level CBOR decoding code.
+#
+# The result of the function is that a library is added to the project which
+# contains the generated decoding code. The code is generated at build time using
+# a script, and generates a c file and an h file with names derived from the
+# name of the <cddl_file>.
+function(cddl_source target cddl_file)
```

Comment:
Should we do a test here, in order to ensure that `cddl_file` actually exists, and is a file.
Could safe some headache later if someone makes a type, and need to trace down the error.